src/Tokenizer/Tokens.php

```diff
@@ -156,19 +156,21 @@ public static function fromArray($array, $saveIndexes = null)
      */
     public static function fromCode($code)
     {
-        $codeHash = self::calculateCodeHash($code);
+        if ('' !== $code) {
```

Comment:
IIRC if you run the tokens with input code `''` and than set other other code things went wrong, I'll hunt this down